node-graph/graph-craft/src/proto.rs

```diff
@@ -491,7 +494,11 @@ impl ProtoNetwork {
 		self.inputs = self.inputs.iter().filter_map(|id| new_positions.get(id).copied()).collect();
 		self.output = *new_positions.get(&self.output).unwrap();
 
-		assert_eq!(order.len(), self.nodes.len());
+		if order.len() != self.nodes.len() {
+			return Err("Assertion failed: order.len() == self.nodes.len()".to_string());
+		}
+
+		return Ok(());
```

Comment:
I think this should actually still be an assert because a failure of this function indicates a bug in our code and not user error right?

Reply:
I think that's right, and we don't need the return on the "Ok()" here either. This has now been fixed.